csfieldguide/chapters/content/en/coding-compression/sections/general-purpose.md

```diff
@@ -43,27 +23,308 @@ Here are some of the bits from the example earlier in this chapter; you can past
+
+```
+I A
+```
+
+Now, recall that the decoder knows that `I + <the next character>` must be the next dictionary entry. The decoder just discovered that the next character is A, so the sequence is:
+
+```
+Sequence  = I + the next character
+	      = I + A
+		  = IA
+```
+
+`IA` is not already in the dictionary, so it is added and assign it to the code 4. The docoder’s dictionary now looks like this:
+
+| Code | Bits         | Sequence |
+|------|--------------|----------|
+| 0    | 000000000000 | I        |
+| 1    | 000000000001 | A        |
+| 2    | 000000000010 | M        |
+| 3    | 000000000011 | S        |
+| 4    | 000000000100 | IA       |
+
+And we continue on to the next few characters in the message:
+
+| Step | Code | Output | New Dict Entry Code | New Dict Sequence | Next Proposed Dict Sequence |
+|------|------|--------|---------------------|-------------------|-----------------------------|
+| 3    | 2    | M      | 5                   | AM                | M?                          |
+| 4    | 3    | S      | 6                   | MS                | S?                          |
+| 5    | 5    | AM     | 7                   | SA                | AM?                         |
+
+In step 5 we have to decode 5. Recall that 5 wasn’t in the initial dictionary, but since we added it in step 2 are able to decode it.
+
+| Step | Code | Output | New Dict Entry Code | New Dict Sequence | Next Proposed Dict Sequence |
+|------|------|--------|---------------------|-------------------|-----------------------------|
+| 6    | 7    | SA     | 8                   | AMS               | SA?                         |
+| 7    | 2    | M      | 9                   | SAM               | M?                          |
+| 8    | 4    | IA     | 10                  | MI                | IA?                         |
+| 9    | 2    | M      | 11                  | IAM               | M?                          |
+
+Once we have decoded the whole message we now have the following string:
+
+```
+IAMSAMSAMIAM
+```
+
+So we successfully encoded and decoded the message!
+And this is the dictionary that the decoder created:
+
+| Code | Bits         | Sequence |
+|------|--------------|----------|
+| 0    | 000000000000 | I        |
+| 1    | 000000000001 | A        |
+| 2    | 000000000010 | M        |
+| 3    | 000000000011 | S        |
+| 4    | 000000000100 | IA       |
+| 5    | 000000000101 | AM       |
+| 6    | 000000000110 | MS       |
+| 7    | 000000000111 | SA       |
+| 8    | 000000001000 | AMS      |
+| 9    | 000000001001 | SAM      |
+| 10   | 000000001010 | MI       |
+| 11   | 000000001011 | IAM      |
+
+Notice that it is exactly the same dictionary as the encoder created - the dictionary has been reconstructed without having to send it to the decoder!
+
+
+{panel type="extra-for-experts" subtitle="true"}
+
+# Extra for Experts
+
+## Other languages
```

Comment:
This is a new way of doing panel labels. We should be consistent - should all the other chapters be altered to use this style? Also I don't think 'extra for experts' fits the content in either case, plus the colon is in the wrong place when this is built. For now I think we should use the old style
![image](https://user-images.githubusercontent.com/25916475/55844507-bb2a2e00-5b91-11e9-9afc-089def6b5999.png)